### PR TITLE
Switch to aeson >= 1

### PR DESCRIPTION
**Pull request by sjakobi**
With this change, the error message for a bad key chord in the config
file changes slightly:

Original error message:

    Could not parse config file /home/simon/.vgrep/config.yaml:
    Aeson exception:
    Error in $.keybindings['results-keybindings']: failed to parse field keybindings: failed to parse field results-keybindings: Unknown key 'bad-chord'
    Falling back to default config.

New error message:

    Could not parse config file /home/simon/.vgrep/config.yaml:
    Aeson exception:
    Error in $.keybindings['results-keybindings']['bad-chord']: Unknown key 'bad-chord'
    Falling back to default config.

### CODE REVIEW

File: src/Vgrep/Key.hs [hunk 1]
```diff
@@ -21,18 +21,33 @@ module Vgrep.Key
...
   )where
 
-import Prelude hiding (Left, Right)
 import           Control.Applicative
+import           Control.Monad
+import           Data.Aeson
+    ( FromJSON (..)
+    , FromJSONKey (..)
+    , FromJSONKeyFunction (..)
+    )
```
Comment (sjakobi) [inline src/Vgrep/Key.hs]: That was `stylish-haskell`, not me… ;)

File: stack.yaml [hunk 2]
```diff
@@ -1,18 +1,14 @@
...
 packages:
 - '.'
 
 # Packages to be pulled from upstream that are not in the resolver (e.g., acme-missiles-0.3)
-extra-deps:
-  # We want a recent version of stylish-haskell.
-  # This is olny for tooling, so does not affect
-  # dependency management.
-  - stylish-haskell-0.6.1.0
+extra-deps: []
```
Comment (sjakobi) [inline stack.yaml]: nightly-2017-01-13 includes stylish-haskell-0.6.5.0

**Comment by sjakobi** [comment]
To address [the build failures](https://travis-ci.org/fmthoma/vgrep/builds/191715665): How I about a `stack-7.10.yaml` to check that vgrep stays compatible with GHC-7.10? A Travis job using stack-7.10.yaml would then replace the current 4 jobs that get `--resolver` `ARGS`.

**Comment by fmthoma** [comment]
@sjakobi Thank you for your PR!

I'm unfortunately a bit torn on bumping the aeson dependency already. I try to support a broad range of versions in order to enable building on different platforms, and a lot of them (most prominently Debian/Ubuntu) are still on 0.11 (or even 0.10). Switching to aeson-1.0 would break `*.deb` packaging.

Ideally, I'd like to support aeson in the range [0.9, 10.0) ∪ [11.0, 1.1), but I'm not sure if this is possible without CPP...

**Comment by sjakobi** [comment]
> Switching to aeson-1.0 would break *.deb packaging.

That's a good point!

> Ideally, I'd like to support aeson in the range [0.9, 10.0) ∪ [11.0, 1.1), but I'm not sure if this is possible without CPP...

Wrapping `Map Key.Chord Command` in a newtype and reusing the old `FromJSON` instance for that type might work. I can make a new PR if you would like to test that approach.

**Comment by sjakobi** [comment]
> I'd like to support aeson in the range [0.9, 10.0) ∪ [11.0, 1.1)

Also, note that [`aeson-1.1.0.0`](https://hackage.haskell.org/package/aeson-1.1.0.0) already exists. WRT compatibility that shouldn't be a problem.

**Comment by fmthoma** [comment]
> Wrapping `Map Key.Chord Command` in a newtype and reusing the old `FromJSON` instance for that type might work. I can make a new PR if you would like to test that approach.

Thank you, that would be very welcome!

> Also, note that aeson-1.1.0.0 already exists. WRT compatibility that shouldn't be a problem.

Right, I haven't checked the changelog yet if there is any breakage.

**Comment by sjakobi** [comment]
Superseded by #39.

**Pull request closed without merging**